src/views/DidUpgradePromo/DidUpgradePromo.tsx

```diff
@@ -0,0 +1,131 @@
+import { useCallback, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { browser } from 'webextension-polyfill-ts';
+
+import * as styles from './DidUpgradePromo.module.css';
+
+import { Identity } from '../../utilities/identities/types';
+import {
+  PasswordField,
+  usePasswordField,
+} from '../../components/PasswordField/PasswordField';
+import { saveIdentity } from '../../utilities/identities/identities';
+import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
+import { TxStatusModal } from '../../components/TxStatusModal/TxStatusModal';
+import { LinkBack } from '../../components/LinkBack/LinkBack';
+import { Stats } from '../../components/Stats/Stats';
+import { paths } from '../paths';
+import {
+  createDid,
+  getDidCreationDetails,
+  getPromoStatus,
+  waitFinalized,
+} from '../../utilities/didUpgradePromo/didUpgradePromo';
+import { useSwrDataOrThrow } from '../../utilities/useSwrDataOrThrow/useSwrDataOrThrow';
+
+interface Props {
+  identity: Identity;
+}
+
+export function DidUpgradePromo({ identity }: Props): JSX.Element | null {
+  const t = browser.i18n.getMessage;
+
+  const [txHash, setTxHash] = useState<string>();
+
+  const [submitting, setSubmitting] = useState(false);
+  const [status, setStatus] = useState<'pending' | 'success' | 'error' | null>(
+    null,
+  );
+
+  const promoStatus = useSwrDataOrThrow('', getPromoStatus, 'getPromoStatus');
+
+  const passwordField = usePasswordField();
+
+  const handleSubmit = useCallback(
+    async (event) => {
+      event.preventDefault();
+
+      try {
+        if (!promoStatus) {
+          throw new Error();
```

Comment:
Your future self, while debugging an issue, will thank you for providing custom messages for errors thrown. In case you’re quite confident this condition will never happen (like on that line) you can just return instead.